launchdarkly-server-sdk/src/data_source.rs

```diff
@@ -89,6 +89,8 @@ impl StreamingDataSource {
                     .delay_max(Duration::from_secs(30))
                     .build(),
             )
+            .connect_timeout(Duration::from_secs(10))
+            .read_timeout(Duration::from_secs(300)) // LaunchDarkly sends keepalives every 3m
```

Comment:
I can imagine that this is causing most of the problems.

When a connection cannot be established or for some reason is dropped, the client tries to re-establish the connection, possibly with an exponential backoff strategy (see where `ReconnectingRequest::poll_next` sets `State::WaitingToReconnect`). With the default `ReconnectOptions` we will be waiting at most 1 minute between reconnect attempts.

However I'm not sure when and how the `read_timeout` will be applied. I suspect that for already established connections that get wedged on some follow-up handshake or for some reason just block on receiving the next event this might really help.